### PR TITLE
[REF] payment: cleanup payment link wizard

### PR DESCRIPTION
**Pull request by chevalierv**
Delete code that is not useful anymore and was introduced in
task-2666881 (PR #78334)

task-2694027

### CODE REVIEW

File: addons/payment/wizards/payment_link_wizard_views.xml [hunk 1]
```diff
@@ -17,9 +17,8 @@
                         <field name="has_multiple_acquirers" invisible="1"/>
                         <field name="description"/>
                         <field name="amount"/>
-                        <field name="acquirer_id"
-                            placeholder="Leave empty to allow all acquirers"
-                            options="{'no_open': True, 'no_create': True}"
+                        <field name="payment_acquirer_selection"
+                            widget="selection"
```
Comment (AntoineVDV) [inline addons/payment/wizards/payment_link_wizard_views.xml:21]: Isn't it already the default?

File: addons/payment/wizards/payment_link_wizard.py [hunk 2]
```diff
@@ -83,22 +47,14 @@ def fields_view_get(self, *args, **kwargs):
...
-        domain="[('id', 'in', available_acquirer_ids)]",
-        help="Force the customer to pay via the specified payment acquirer. Leave empty to allow the customer to choose among all acquirers."
-    )
     has_multiple_acquirers = fields.Boolean(
         string="Has Multiple Acquirers",
         compute='_compute_has_multiple_acquirers',
     )
     payment_acquirer_selection = fields.Selection(
-        string="Payment acquirer selected",
+        string="Force Payment Acquirer",
```
Comment (AntoineVDV) [inline addons/payment/wizards/payment_link_wizard.py:55]:
```suggestion
        string="Allow Payment Acquirer",
        help="If a specific payment acquirer is selected, customers will only be allowed to pay "
             "via this one. If 'All' is selected, customers can pay via any available payment "
             "acquirer.",
```

File: addons/payment/wizards/payment_link_wizard_views.xml (hunk 1 above)
Comment (chevalierv) [reply addons/payment/wizards/payment_link_wizard_views.xml:21]: Yes indeed, I removed it after testing that it didn't have any impact.

**Review by AntoineVDV — APPROVED** [review]
@robodoo r+

**Pull request closed without merging**